### ISSUE TITLE
Standard SKU Loadbalancer does not use own static public ip as outbound IP

### ISSUE THREAD

**Issue by angrox** [Azure/AKS#1314]
I want to create an AKS cluster with the Standard SKU Loadbalancer. I have created a public static ip (SKU Standard) in another resourcegroup and want to bind the static IP address to the AKS loadbalancer as incoming AND outgoing ip address-

**What happened**:
Azure AKS Standard Loadbalancer creates managed IP and uses this as outgoing IP. 

**What you expected to happen**:
I want my own public IP as outgoing IP

**How to reproduce it (as minimally and precisely as possible)**:
* Install AKS with Standard SKU loadbalancer
* Deploy NGINX helm chart with static public ip (documentation: https://docs.microsoft.com/en-us/azure/aks/ingress-static-ip)
* Connect to a pod and check your outgoing IP address - it does not match your static public ip

**Anything else we need to know?**:
-

**Environment**:
- Kubernetes version 1.14.8
NGINX helm chart version: nginx-ingress-1.24.7
- Nodes: 5

**Comment by angrox** [comment]
I tried the following:

* Set managed IPs to zero: az aks upgrade -g <rg> -n <name> --load-balancer-managed-outbound-ip-count 0
-> does not work. No errors, just no effect. Managed IP still there. 

* Deinstall nginx and publish my public ip as lb ip:
az aks update -g <rg> -n <name> --load-balancer-outbound-ips <public ip address path>
-> Sets my ip as first ip address in the lb. When I try to install nginx the loadbalancer cannot be ensured: 

```
Events:
  Type     Reason                      Age                  From                Message
  ----     ------                      ----                 ----                -------
  Normal   EnsuringLoadBalancer        59s (x7 over 6m16s)  service-controller  Ensuring load balancer
  Warning  CreatingLoadBalancerFailed  59s (x7 over 6m16s)  service-controller  Error creating load balancer (will retry): failed to ensure load balancer for service nginx-ingress/nginx: timed out waiting for the condition
```

**Comment by palma21** [comment]
So a few things:

If you want to use your already existing IP for outbound your second command should be correct.
`az aks update -g rg -n name --load-balancer-outbound-ips public_ip_address_path`

If by address path you mean the IP resource ID, but by your description it seemed to work correctly.

Then you install nginx and you want that IP to also be the Inbound IP for your Nginx per your OP so I'm guessing you're specifying it on the nginx service correctly, something like: https://docs.microsoft.com/en-us/azure/aks/static-ip#create-a-service-using-the-static-ip-address

Per your description if nothing else was done, what might be happening is that your kubernetes cluster does not have access to your Public IP in its separate RG. When you do the AKS update command it's you that technically are assigning that IP to the LB but when you specify it for inbound on the k8s service it's kubernetes that is going to try and get the IP, so in the service you need to specify the RG of the IP and you need to give permissions to your kubernetes service principal (created when the cluster is created) which is the cluster identity, like on the link I pasted above.

**Comment by xuto2** [comment]
Agree with what @palma21's analysis above. If you still have the issue you can file a ticket to us and we can help confirm the issues.

**Comment by brennerm** [comment]
I'm having the same issue. Assigning the public IP to the AKS Standard Load Balancer and setting it in the nginx configuration results in the following error:

```
ingress        1s          Warning   CreateOrUpdateLoadBalancer   service/ingress-service-nginx-ingress-controller
                  network.LoadBalancersClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="PublicIPReferencedByMultipleIPConfigs" Message="Public ip address /subscriptions/xxxxxxxxxxxxxxxxxxxxxxxxxxxxx/resourceGroups/prod-ips/providers/Microsoft.Network/publicIPAddresses/aks-ip is referenced by multiple ipconfigs in resource /subscriptions/xxxxxxxxxxxxxxxxxxxxxxxxxxxxx/resourceGroups/aks/providers/Microsoft.Network/loadBalancers/kubernetes." Details=[]
ingress        1s          Warning   CreatingLoadBalancerFailed   service/ingress-service-nginx-ingress-controller
                  Error creating load balancer (will retry): failed to ensure load balancer for service ingress/ingress-service-nginx-ingress-controller: timed out waiting for the condition
```

A similar setup with a Basic LB works just fine as I don't have to configure the outbound IP when creating the AKS.

**Comment by brennerm** [comment]
Current workaround is to let Azure create a public IP address and to let ingress-nginx create a second Frontend IP configuration in the LB. Still requires me to pay for two Public IP addresses instead of one.

**Comment by palma21** [comment]
So you're trying to associate the existing egress IP and re-use it for ingress. But when you reference it's not able to do it. 

can you share your svc yaml file?

**Comment by brennerm** [comment]
@palma21 I'm using the nginx-ingress Helm chart. So my setup is very close to the one described here: https://docs.microsoft.com/en-us/azure/aks/ingress-static-ip

**Comment by ekarlso** [comment]
Same for me/us, typically end up paying 2x the amount of ip's.

**Comment by slynickel** [comment]
If I'm reading the change log right this should be fixed with the most recent AKS update. I'm never actually sure when the rollout occurs though (no way to know when a region gets it as far as I know).

https://github.com/Azure/AKS/blob/master/CHANGELOG.md#release-2020-07-06

>* Features
>
>   * Users can now reuse inbound and outbound IPs on the Load Balancer. After this change, users can assign an outbound IP that is same as an inbound IP in the AKS SLB.

The wording of that last sentence seems problematic though. Is it the case that you can _only_ assign an outbound IP address that is already an existing inbound IP address? Or can you assign IPs to be outbound then also make them inbound IP addresses?

**Comment by palma21** [comment]
The former works after this release. The latter requires upstream fixes to cloud provider but it's being worked on as well.
https://github.com/kubernetes-sigs/cloud-provider-azure/issues/267

**Comment by palma21** [comment]
Converting this into a feature request pending the completion of the above item.

**Comment by swapnilpotnis** [comment]
> So a few things:
> 
> If you want to use your already existing IP for outbound your second command should be correct.
> `az aks update -g rg -n name --load-balancer-outbound-ips public_ip_address_path`
> 
> If by address path you mean the IP resource ID, but by your description it seemed to work correctly.
> 
> Then you install nginx and you want that IP to also be the Inbound IP for your Nginx per your OP so I'm guessing you're specifying it on the nginx service correctly, something like: https://docs.microsoft.com/en-us/azure/aks/static-ip#create-a-service-using-the-static-ip-address
> 
> Per your description if nothing else was done, what might be happening is that your kubernetes cluster does not have access to your Public IP in its separate RG. When you do the AKS update command it's you that technically are assigning that IP to the LB but when you specify it for inbound on the k8s service it's kubernetes that is going to try and get the IP, so in the service you need to specify the RG of the IP and you need to give permissions to your kubernetes service principal (created when the cluster is created) which is the cluster identity, like on the link I pasted above.

Re: With reference to the command "az aks update -g rg -n name --load-balancer-outbound-ips public_ip_address_path", what is understand is I would be able to set the outbound IP of AKS to the one which I have currently assigned to my service. But what if I have another service (for.eg SERVICE_B) with a Public IP(fore.g PUB_IP_2) assigned Load Balancer, and I wish the traffic for SERVICE_B to always to be routed in and out using PUB_IP_2.
As far as I understand, AKS would always use the first public IP(for.e.g PUB_IP_1) which has been assigned to it.

In short, how I ensure that the following takes place:
Client <-> PUB_IP_2 of AKS <-> SERVICE_2
Client <-> PUB_IP_1 of AKS <-> SERVICE_1
